### ISSUE TITLE
libtype specification

### ISSUE THREAD

**Issue by gresteban** [COMBINE-lab/salmon#82]
Version 0.7.1 seems to refer to "SF" libtype as just "F". Maybe this is an abbreviation done on purpose since "F" alone necessarily implies "stranded" but I think that can be confusing. Moreover, in lib_format_counts.json file, both "F" and "SF" are listed, but only "F" gets counts whereas "SF" is 0. "F" is not recognized as a valid library type for the parameter `-l`, only "SF" is (as expected).

I haven't tested it but the same probably happens with "R" and "SR".

**Comment by rob-p** [comment]
Hi @gresteban,

  Thanks for reporting this.  Did this happen when running with the automatic library type detection or with `-l SF`?  It is *not* intended, but knowing what code path was executed will help me narrow down the source of the bug.

**Comment by gresteban** [comment]
Hi and thanks for your (as usual) rapid response.

It happens with both `-l A` and `-l SF`. In both cases libraries are reported as "F".

The command line that I used was pretty straight forward:

`salmon quant -i ./salmon_idx -l A   -r sample.fastq.gz  -o salmon_quant_01;`

`salmon quant -i ./salmon_idx -l SF -r sample.fastq.gz  -o salmon_quant_02;`

The content of the file lib_format_counts.json is the same in both cases:

```
{
    "read_files": "sample.fastq.gz",
    "expected_format": "F",
    "compatible_fragment_ratio": 0.98134531696722038,
    "num_compatible_fragments": 628482,
    "num_assigned_fragments": 640429,
    "num_consistent_mappings": 748760,
    "num_inconsistent_mappings": 16217,
    "MSF": 0,
    "OSF": 0,
    "ISF": 0,
    "SF": 0,
    "MSR": 0,
    "OSR": 0,
    "ISR": 0,
    "SR": 0,
    "MF": 0,
    "OF": 0,
    "IF": 0,
    "F": 748760,
    "MR": 0,
    "OR": 0,
    "IR": 0,
    "R": 16217,
    "MU": 0,
    "OU": 0,
    "IU": 0,
    "U": 0
}
```

**Comment by rob-p** [comment]
Hi @gresteban,

  Thanks for reporting this.  One issue is that the library types are kept encoded in a dense table.  Some codes correspond to "invalid" library types (i.e. types we'd never see in practice like single end, with a paired-end strandedness type like forward-reverse, etc.).  Even those these never garner counts, the `lib_format_counts.json` file was generated by enumerating all possible library type codes, which is why you see more than 12 different library formats even though there are only 12 valid formats.  The other issue was that, though the encoding was right internally, I was writing out single end, stranded, forward reads as 'F' and single end, stranded, reverse reads as 'R' when these should be 'SF' and 'SR'.  This issue should be addressed by commit a944f7d, and the fix will make it into the next release of Salmon.  Thanks again!

**Comment by gresteban** [comment]
Great!

Thanks for your work.
